### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by bmatherly**
producer->profile returns a new object that is never freed

### CODE REVIEW

File: src/dialogs/filedatedialog.cpp [hunk 1]
```diff
@@ -113,7 +115,8 @@ void FileDateDialog::populateDateOptions(Mlt::Producer *producer)
     }
 
     // Add metadata dates
-    Mlt::Producer tmpProducer( *(producer->profile()), "avformat", resource.toUtf8().constData() );
+    Mlt::Profile profile(mlt_profile_clone(producer->get_profile()));
+    Mlt::Producer tmpProducer( profile, "avformat", resource.toUtf8().constData() );
```
Comment (ddennedy) [inline src/dialogs/filedatedialog.cpp:119]: Why not use `MLT.profile()` here like we do so many other places?
There is not supposed to be spaces inside the parenthesis.

**Comment by ddennedy** [comment]
`TextProducerWidget` also has a place where it should be using `MLT.profile()`:
```
--- src/widgets/textproducerwidget.cpp
+++ src/widgets/textproducerwidget.cpp
@@ -146,8 +146,7 @@ void TextProducerWidget::loadPreset(Mlt::Properties &p)
         filter.reset(MLT.getFilter(kRichFilterName, m_producer.data()));
         if (filter && filter->is_valid())
             m_producer->detach(*filter);
-        Mlt::Profile profile = m_producer->get_profile();
-        filter.reset(createFilter(profile, m_producer.data()));
+        filter.reset(createFilter(MLT.profile(), m_producer.data()));
         m_producer->attach(*filter);
         emit producerChanged(m_producer.data());
     }
```
This is the only other problematic usage of `Mlt::Service::profile()` I found.

File: src/dialogs/filedatedialog.cpp (hunk 1 above)
Comment (bmatherly) [reply src/dialogs/filedatedialog.cpp]: Not a compelling reason - but so much depends on mltcontroller.h. If I change one small thing in mltcontroller.h, the whole application has to recompile. I think we should prefer MLT.profile() over Mlt::Service::profile() in all cases.

Comment (bmatherly) [reply src/dialogs/filedatedialog.cpp]: > There is not supposed to be spaces inside the parenthesis.

done

**Comment by bmatherly** [comment]
> TextProducerWidget also has a place where it should be using MLT.profile()

Done

**Pull request merged**